thriftpy/thrift.py

```diff
@@ -189,19 +190,19 @@ def process_in(self, iprot):
             iprot.skip(TType.STRUCT)
             iprot.read_message_end()
             return api, seqid, TApplicationException(TApplicationException.UNKNOWN_METHOD), None   # noqa
-        else:
-            args = getattr(self._service, api + "_args")()
-            args.read(iprot)
-            iprot.read_message_end()
-            result = getattr(self._service, api + "_result")()
 
-            # convert kwargs to args
-            api_args = [args.thrift_spec[k][1]
-                        for k in sorted(args.thrift_spec)]
-            call = lambda: getattr(self._handler, api)(
-                *(args.__dict__[k] for k in api_args)
-            )
-            return api, seqid, result, call
+        args = getattr(self._service, api + "_args")()
+        args.read(iprot)
+        iprot.read_message_end()
+        result = getattr(self._service, api + "_result")()
+
+        # convert kwargs to args
+        api_args = [args.thrift_spec[k][1]
+                    for k in sorted(args.thrift_spec)]
+        call = lambda: getattr(self._handler, api)(
```

Comment:
There's also a E731 "do not assign a lambda expression, use a def" violation in here.

I've updated the code in develop branch to use the `def`.